### PR TITLE
Add unit tests to percentile ranks aggregations.

### PR DESCRIPTION
**Pull request by jpountz**
Relates #22278

### CODE REVIEW

File: test/framework/src/main/java/org/elasticsearch/test/AbstractWireSerializingTestCase.java [hunk 1]
```diff
@@ -61,10 +61,6 @@ public void testEqualsAndHashcode() throws IOException {
 
             T secondInstance = copyInstance(firstInstance);
             assertEquals("instance is not equal to self", secondInstance, secondInstance);
-            if (false == firstInstance.equals(secondInstance)) {
-                firstInstance.equals(secondInstance);
-            assertEquals("instance is not equal to its copy", firstInstance, secondInstance);
-            }
```
Comment (jimczi) [inline test/framework/src/main/java/org/elasticsearch/test/AbstractWireSerializingTestCase.java]: Why do you need to remove this assertion ?

**Review by jimczi — APPROVED** [review]
Left a small comment regarding the removal of an assertion.
LGTM otherwise

File: test/framework/src/main/java/org/elasticsearch/test/AbstractWireSerializingTestCase.java (hunk 1 above)
Comment (jpountz) [inline test/framework/src/main/java/org/elasticsearch/test/AbstractWireSerializingTestCase.java]: Good catch, I wanted only to remove what looked like a debug statement, I pushed a fix.

Comment (jimczi) [inline test/framework/src/main/java/org/elasticsearch/test/AbstractWireSerializingTestCase.java]: Thanks, LGTM

**Pull request merged**